internal/guest/bridge/publisher.go

```diff
@@ -0,0 +1,80 @@
+//go:build linux
+// +build linux
+
+package bridge
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/guest/prot"
+	log "github.com/Microsoft/hcsshim/internal/log"
+	eventstypes "github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/events"
+	"github.com/containerd/containerd/runtime"
+	"github.com/containerd/containerd/runtime/v2/shim"
+	"github.com/pkg/errors"
+)
+
+const emptyActivityID = "00000000-0000-0000-0000-000000000000"
+
+var ErrBridgePublisherClosed = errors.New("bridge publisher closed")
+
+// wraps `bridge.PublishNotification` to conform to `shim.Publish`, so that it can
+// be used by containerd's `oom.Watcher` (github.com/containerd/containerd/pkg/oom)
+type BridgePublisher struct {
+	f func(*prot.ContainerNotification)
+}
+
+var _ shim.Publisher = BridgePublisher{}
+
+func NewBridgePublisher(publisherFunc func(*prot.ContainerNotification)) BridgePublisher {
```

Comment:
This shouldn't need to be exported, it's only used internally to this package.

Reply:
Probably some of this other stuff doesn't need to be exported either. Worth checking on.